src/edu/psu/sweng500/team8/play/GameSession.java

```diff
@@ -44,22 +45,31 @@ public Board getGameBoard() {
 	}
 	
 	/**
-	 * Undo last action such as entering a number or redo
+	 * Undo last action such as entering a number, deleting a number or redo
+	 * 
 	 */
 	public void doUndo() {
 		/* revert last action on the sudokuActionQueue, and put that action into sudokuActionQueueForUndo */
 		if (!sudokuActionQueue.isEmpty()) {
 			SudokuAction lastAction = sudokuActionQueue.pop();
+		
+			PencilMarkManager previousPencilMarkManager = lastAction.getPencilMarkManager();
+			/* TODO - we need to get the current Pencil Mark state */
+			PencilMarkManager currentPencilMarkManager = null;
+			lastAction.setPencilMarkManager(currentPencilMarkManager);

```

Comment:
There should only be one pencil mark manager for the board. I don't think it's necessary per-action. Also, why are you setting it null here?